### PR TITLE
MOTECH-1869: Make creating an index for a lookup optional

### PR DESCRIPTION
**Pull request by shubhambeehyv**
Added indexRequired field in Lookup to make indexing optional.

### CODE REVIEW

**Comment by shubhambeehyv** [comment]
test this please

File: platform/mds/mds/src/main/java/org/motechproject/mds/annotations/internal/LookupProcessor.java [hunk 1]
```diff
@@ -113,6 +114,7 @@ protected void process(AnnotatedElement annotatedElement) {
         lookup.setLookupFields(lookupFields);
         lookup.setReadOnly(true);
         lookup.setMethodName(method.getName());
+        lookup.setIndexRequired(indexRequired);
```
Comment (pgesek) [inline platform/mds/mds/src/main/java/org/motechproject/mds/annotations/internal/LookupProcessor.java]: we should  extend the lookup processor test to check this

**Comment by 1stmateusz** [comment]
test this please

**Comment by pgesek** [comment]
Moved to #327

**Pull request closed without merging**